react/src/components/atoms/table/Table/index.js

```diff
@@ -11,68 +12,84 @@ const Table = (table) => {
                     >{cell.text}
                     </th>
-                  ) : (
-                    <td
-                      key={cell.key || defaultCellKey}
-                      data-label={dataLabel}
-                      colSpan={cell.colspan || null}
-                      rowSpan={cell.rowspan || null}
-                    >
-                      {cell.text}
-                    </td>
-                  ));
-              })}
-              </tr>);
-          })}
-        </tbody>
-        ))}
-    </table>
+                ))}
+                </tr>
+              ))}
+            </thead>
+        }{table.bodies.map((body, bodyIndex) => (
+          <tbody key={body.key || `tbody${bodyIndex}`}>
+            { body.rows.map((row, rowIndex) => {
+              const tableIndexOffset = row.rowSpanOffset ? 0 : 1;
+              const defaultRowKey = `tbody${bodyIndex}-row${rowIndex}`;
+              return(
+                <tr key={row.key || defaultRowKey} className={row.rowSpanOffset ? 'is-offset' : null}>
+                  { row.cells.map((cell, cellIndex) => {
+                    let dataLabel = null;
+                    if ((cellIndex > 0) && tableHeaderRow.cells && tableHeaderRow.cells.length > 0) {
+                      // Mayflower twig loops started at 1 - add one to index.
+                      dataLabel = tableHeaderRow.cells[(cellIndex + 1) - tableIndexOffset].text;
+                    }
+                    const defaultCellKey = `${defaultRowKey}-cell${cellIndex}`;
+                    const cellClasses = classnames({
+                      'ma__table-cell': true,
+                      'no-left-pad': (!row.rowSpanOffset && !cell.heading && cellIndex === 1)
+                    });
+                    return(cell.heading ? (
+                      <th
+                        className={cellClasses}
+                        key={cell.key || defaultCellKey}
+                        data-label={dataLabel}
+                        scope="row"
+                        colSpan={cell.colspan || null}
+                        rowSpan={cell.rowspan || null}
+                      >{cell.text}
+                      </th>
+                    ) : (
+                      <td
+                        className={cellClasses}
+                        key={cell.key || defaultCellKey}
+                        data-label={dataLabel}
+                        colSpan={cell.colspan || null}
+                        rowSpan={cell.rowspan || null}
+                      >
+                        {cell.text}
+                      </td>
+                    ));
+                })}
+                </tr>
+              );
+            })
+            }
+          </tbody>))
+          }
+        </table>
+      </div>
+    </div>
   );
 };
 
 Table.propTypes = {
+  id: PropTypes.string,
   head: PropTypes.shape({
     rows: PropTypes.arrayOf(PropTypes.shape({
       rowSpanOffset: PropTypes.bool,
```

Comment:
can you add a description here for the rowSpanOffset. Currently throwing an error if true in bodies.rows, and I don't think it's doing anything in head.rows

Reply:
The initial example already uses it with it some things being true/false, and there's no error.